### ISSUE TITLE
mod_muc not sending presence when revoking affiliation if the revoker is not connected to the room

### ISSUE THREAD

**Issue by mrfrostfire** [processone/ejabberd#1135]
> What version of ejabberd are you using?

16.02

> What operating system (version) are you using?

Red Hat

> How did you install ejabberd (source, package, distribution)?

package

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

I have a client bot that takes care of creating, modifying, setting affiliation etc to moderated member-only muc rooms.

Creating rooms etc works, setting affiliation too, except when it comes to revoking affiliation. If I set the affiliation to none for a user, the affiliation is set and the server responds with a success result, but the presence that the user has been revoked and thereby kicked out is never sent even if the user being revoked is still in the room.

If I join the room with my bot prior to revoking the users affiliation, the presence is sent properly.

**Comment by badlop** [comment]
Problem confirmed. In a member-only room, if the admin that sends this stanza is not inside the room, then user2 does not receive the kick stanza.
```
<iq type='set' to='room1@conference.localhost'>
  <query xmlns='http://jabber.org/protocol/muc#admin'>
    <item nick='user2'
	affiliation='none'/>
  </query>
</iq>
```

**Comment by badlop** [comment]
Fix problem in dfee843 when non-occupant admin kicks an occupant (#1135)
